privateSBOMExchange/src/petra/lib/models/tree_ops.py

```diff
@@ -172,11 +184,15 @@ def get_membership_proof(root, target_hash):
     def get_prefix(node):
         """This is same prefix used by MerkleVisitor during hashing"""
         if isinstance(node, SbomNode):
-            prefix = (f"{node.purl}").encode() 
+            prefix = node.redacted_keys+(f"{node.purl}").encode("utf-8") +node.plaintext_hash 
         elif isinstance(node, ComplexNode):
-            prefix = (f"{node.encrypted_data}{node.complex_type}{node.policy}").encode()
+            prefix = (f"{node.encrypted_data}{node.policy}{node.complex_type}").encode("utf-8")+node.plaintext_commit.value+node.plaintext_hash
```

Comment:
why are we doing `+` here? wouldn't it be better to put it on the format string ?

Reply:
ComplexNodes now support `serialize_for_hashing()` so we ensure that we're always hashing the node contents in a consistent order and format. Please call that function here instead.

Reply:
@SantiagoTorres  can I? they are of different types
@marcelamelara  I can't use the serialization here, because it appends the children

Reply:
I believe the serialization for the ComplexNode takes the children_hashes as a parameter, so you should be able to pass in an empty buffer if you don't want to append the children.

Reply:
oh yes ! I  just pushed the requested changes.